app/api/v1/clan_icons.py

```diff
@@ -0,0 +1,77 @@
+import logging
+
+from fastapi import APIRouter
+from fastapi import Depends
+from fastapi import File
+from fastapi import Response
+
+import app.usecases.images
+import settings
+from app.adapters import s3
+from app.api.authorization import AdminAuthorization
+from app.api.authorization import authorize_admin
+from app.errors import Error
+from app.errors import ErrorCode
+from app.usecases.images import ImageType
+
+router = APIRouter()
+
+
+def _get_status_code_for_error(error_code: ErrorCode) -> int:
+    try:
+        return {
+            ErrorCode.INVALID_CONTENT: 400,
+            ErrorCode.INAPPROPRIATE_CONTENT: 400,
+            ErrorCode.SERVICE_UNAVAILABLE: 503,
+        }[error_code]
+    except KeyError:
+        logging.warning(
+            "Unmapped error code while resolving http status code",
+            extra={"error_code": error_code},
+        )
+        return 500
+
+
+@router.post("/api/v1/clans/{clan_id}/icon")
+async def upload_avatar(
+    clan_id: str,
+    file_content: bytes = File(...),
+    authorization: AdminAuthorization = Depends(authorize_admin),
+):
+    data = await app.usecases.images.upload_image(
+        image_type=ImageType.CLAN_ICON,
+        image_content=file_content,
+        no_ext_file_name=f"{clan_id}",
+        authorization=authorization,
+    )
+    if isinstance(data, Error):
+        return Response(
+            status_code=_get_status_code_for_error(data.code),
+            content=data.user_feedback,
+        )
+    return Response(status_code=201)
+
+
+@router.get("/api/v1/clan-icons/{file_path:path}")
+@router.get("/public/api/v1/clan-icons/{file_path:path}")
+async def get_avatar(file_path: str):
+    if ".." in file_path or "/" in file_path:
+        return Response(status_code=404)
+
+    if not file_path.endswith(".png"):
+        file_path += ".png"
+
+    download_response = await s3.download(file_path, "clan-icons")
```

Comment:
maybe smth like this?
```suggestion
    download_response = await s3.download(file_path, ImageType.CLAN_ICON.get_s3_folder())
```

Reply:
maybe -- as a minimum at least use kwargs for now 
```suggestion
    download_response = await s3.download(file_path, directory="clan-icons")
```